service_test.go

```diff
@@ -9,372 +9,239 @@ import (
+		}
+		got, err := servicesHandler.Update(ctx, "test-pr", "test-sr", UpdateServiceRequest{
+			Cloud:       "google-europe-west1",
+			GroupName:   "default",
+			Plan:        "hobbyist",
+			DiskSpaceMB: 100,
 		})
-	}
+		if err != nil {
+			t.Errorf("Unexpected error was returned when getting service: %v", err)
+			return
+		}
+		if !reflect.DeepEqual(got, expectedResponse) {
+			t.Errorf("Got %v, expected %v", got, expectedResponse)
+		}
+	})
 }
 
 func TestServicesHandler_Delete(t *testing.T) {
 	ctx := context.Background()
-	c, _ := setupServiceTestCase(t)
 
-	type fields struct {
-		client *Client
-	}
-	type args struct {
-		project string
-		service string
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		wantErr bool
-	}{
-		{
-			name:   "",
-			fields: fields{client: c},
-			args: args{
-				project: "test-pr",
-				service: "test-sr",
-			},
-			wantErr: false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			h := &ServicesHandler{
-				client: tt.fields.client,
-			}
-			if err := h.Delete(ctx, tt.args.project, tt.args.service); (err != nil) != tt.wantErr {
-				t.Errorf("Delete() error = %v, wantErr %v", err, tt.wantErr)
-			}
+	t.Run("deletes service", func(t *testing.T) {
+		testServer := setupTestServer(t, "/project/test-pr/service/test-sr", "DELETE", nil)
+
+		t.Cleanup(func() {
+			testServer.Close()
 		})
-	}
+
+		if err := servicesHandler.Delete(ctx, "test-pr", "test-sr"); err != nil {
+			t.Errorf("Unexpected error was returned when getting service: %v", err)
+		}
+	})
 }
 
 func TestServicesHandler_List(t *testing.T) {
 	ctx := context.Background()
-	c, _ := setupServiceTestCase(t)
 
-	type fields struct {
-		client *Client
-	}
-	type args struct {
-		project string
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		want    []*Service
-		wantErr bool
-	}{
-		{
-			name:    "",
-			fields:  fields{client: c},
-			args:    args{project: "test-pr-list"},
-			want:    nil,
-			wantErr: false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			h := &ServicesHandler{
-				client: tt.fields.client,
-			}
-			got, err := h.List(ctx, tt.args.project)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("List() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("List() got = %v, want %v", got, tt.want)
-			}
+	t.Run("returns list of services", func(t *testing.T) {
+		testServer := setupTestServer(t, "/project/test-pr/service", "GET", ServiceListResponse{
+			APIResponse: APIResponse{},
+			Services:    []*Service{},
 		})
-	}
+
+		t.Cleanup(func() {
+			testServer.Close()
+		})
+		got, err := servicesHandler.List(ctx, "test-pr")
+
+		if err != nil {
+			t.Errorf("Unexpected error was returned when listing services: %v", err)
+			return
+		}
+		if !reflect.DeepEqual(got, []*Service{}) {
+			t.Errorf("Got %v, expected %v", got, []*Service{})
+		}
+	})
```

Comment:
Please do not refactor the client. It will be replaced soon with code-generated one.